Extensions/xkit_preferences.js

```diff
@@ -1266,33 +1266,20 @@ XKit.extensions.xkit_preferences = new Object({
 
 			m_html = m_html + '<div id="xkit-extension-panel-no-settings">No settings available for this extension.</div>';
 
-		} else {
+		} else if (XKit.installed.enabled(extension_id) === false) {
 
-			// To-Do: Load Extension settings Here!
-			// Check if custom control panel:
-			if (typeof XKit.extensions[extension_id].cpanel === "undefined") {
-				// Yes it is.
+			m_html = m_html + '<div id="xkit-extension-panel-no-settings">Please enable this extension to customize it.</div>'; 
+			
+		} else {
+			//Get standard settings.
+			if (typeof XKit.extensions[extension_id].preferences !== "undefined") {
 				m_html = m_html + '<div id="xkit-extension-panel-settings">' + XKit.extensions.xkit_preferences.return_extension_settings(extension_id) + "</div>";
-			} else {
-				// Check if it also has standard options:
-				if (XKit.installed.enabled(extension_id) === false) {
-
-					m_html = m_html + '<div id="xkit-extension-panel-no-settings">Please enable this extension to customize it.</div>';
-
-				} else {
-					if (typeof XKit.extensions[extension_id].preferences !== "undefined") {
-						m_html = m_html + '<div id="xkit-extension-panel-settings">' + XKit.extensions.xkit_preferences.return_extension_settings(extension_id) + "</div>";
-					} else {
-						m_html = m_html + '<div id="xkit-extension-panel-settings"><div style="padding: 10px">There is a problem loading the extension panel.<br/>Update the extension and again later.</div></div>';
-					}
-				}
+				
 			}
 		}
-
 		$("#xkit-extensions-panel-right-inner").html(m_html);
-
 		// Pass control to the extension to draw custom control panel:
-		if (typeof XKit.extensions[extension_id].cpanel !== "undefined") {
+		if (typeof XKit.extensions[extension_id].cpanel !== "undefined" && XKit.installed.enabled(extension_id) !== false) {
```

Comment:
I think this PR can be simplified to just this line. XKit's current behavior should already say to enable the extension and the PR as written drops the logic of line 1283/1286 which is a good fallback to have (I think it may be unnecessary but I have no idea)

Reply:
Currently it will only say to enable extension if there's a custom panel, and even then it still draws the actual panel internally, just doesn't find the div to attach it to. If there are only normal preferences it will load them without checking. This whole thing was written weirdly and the comments were misleading, plus the fallback was impossible to trigger sans cosmic rays flipping bits so I decided to rewrite it in a more streamlined fashion. I really don't think it should be simplified to just two lines.
With that said, there's one edge case I missed here, I'm gonna add it now.